lsql/judge/tests/test_hint.py

```diff
@@ -167,3 +182,141 @@ def test_show_hints(self):
+        self.assertEqual(hints_expected1[1], hints[0].text_md)
+        self.assertEqual(hints_expected2[0], hints[1].num_submit)
+        self.assertEqual(hints_expected2[1], hints[1].text_md)
+        self.assertEqual(hints_expected3[0], hints[2].num_submit)
+        self.assertEqual(hints_expected3[1], hints[2].text_md)
+
+        # Check hints loaded for FunctionProblem
+        function.clean()
+        function.collection = collection
+        function.save()
+        hints = Hint.objects.filter(problem=function).order_by('num_submit')
+        self.assertEqual(hints.count(), 3)
+        self.assertEqual(hints_expected1[0], hints[0].num_submit)
+        self.assertEqual(hints_expected1[1], hints[0].text_md)
+        self.assertEqual(hints_expected2[0], hints[1].num_submit)
+        self.assertEqual(hints_expected2[1], hints[1].text_md)
+        self.assertEqual(hints_expected3[0], hints[2].num_submit)
+        self.assertEqual(hints_expected3[1], hints[2].text_md)
+
+        # Check hints loaded for ProcProblem
+        proc.clean()
+        proc.save()
+        hints = Hint.objects.filter(problem=proc).order_by('num_submit')
+        self.assertEqual(hints.count(), 3)
+        self.assertEqual(hints_expected1[0], hints[0].num_submit)
+        self.assertEqual(hints_expected1[1], hints[0].text_md)
+        self.assertEqual(hints_expected2[0], hints[1].num_submit)
+        self.assertEqual(hints_expected2[1], hints[1].text_md)
+        self.assertEqual(hints_expected3[0], hints[2].num_submit)
+        self.assertEqual(hints_expected3[1], hints[2].text_md)
+
+        # Check hints loaded for TriggerProblem
+        trigger.clean()
+        trigger.save()
+        hints = Hint.objects.filter(problem=trigger).order_by('num_submit')
+        self.assertEqual(hints.count(), 3)
+        self.assertEqual(hints_expected1[0], hints[0].num_submit)
+        self.assertEqual(hints_expected1[1], hints[0].text_md)
+        self.assertEqual(hints_expected2[0], hints[1].num_submit)
+        self.assertEqual(hints_expected2[1], hints[1].text_md)
+        self.assertEqual(hints_expected3[0], hints[2].num_submit)
+        self.assertEqual(hints_expected3[1], hints[2].text_md)
+
+        # Check hints loaded for DiscriminantProblem
+        discriminant.clean()
+        discriminant.save()
+        hints = Hint.objects.filter(problem=discriminant).order_by('num_submit')
+        self.assertEqual(hints.count(), 3)
+        self.assertEqual(hints_expected1[0], hints[0].num_submit)
+        self.assertEqual(hints_expected1[1], hints[0].text_md)
+        self.assertEqual(hints_expected2[0], hints[1].num_submit)
+        self.assertEqual(hints_expected2[1], hints[1].text_md)
+        self.assertEqual(hints_expected3[0], hints[2].num_submit)
+        self.assertEqual(hints_expected3[1], hints[2].text_md)
+
+    def test_to_long_hint(self):
+        """Test to check if hints.md is loaded correctly fro SelectProblem"""
```

Comment:
`""" Test to check if hints.md is loaded correctly to a SelectProblem. It checks both hints with one line and hints with several lines """`